### PR TITLE
Add warning when artifact store cannot be loaded

### PR DESCRIPTION
**Pull request by strickvl**
Added:

- a more verbose warning when trying to import an artifact store flavor without all dependencies installed
- a boolean flag to specify that you want artifacts loading using only the current stack's artifact store

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/stacks-and-components/component-guide) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.
- [x] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: src/zenml/utils/artifact_utils.py [hunk 1]
```diff
@@ -114,11 +114,17 @@ def load_model_from_metadata(model_uri: str) -> Any:
...
+def load_artifact(
+    artifact: "ArtifactResponseModel",
+    use_current_stack_only: bool = False,
+) -> Any:
     """Load the given artifact into memory.
 
     Args:
         artifact: The artifact to load.
+        use_current_stack_only: If True, only the current stack will be used to
+            load the artifact. If False, the artifact store will be loaded from
```
Comment (schustmi) [inline src/zenml/utils/artifact_utils.py:126]:
```suggestion
            load the artifact. If False, the artifact will be loaded from
```

**Review by schustmi — CHANGES_REQUESTED** [review]
I think I misremembered this function when talking about it on discord. I thought this was some kind of utility function which takes as input a `run_name`, `step_name` and `output_name` and then fetches/loads the artifact.

If this is really just loading a specific artifact then it seems like a user error to try loading it when the integrations aren't available. Maybe in that case my suggestion of having this boolean flag does not make much sense, and we should rather just go with the improved error message that you already implemented as part of this PR as well?

**Comment by strickvl** [comment]
@schustmi the improved error message doesn't help with the fact that I might not want it to assume to use an artifact store other than the one in my current stack. (Agreed though that not having the right dependencies installed is just user error which we can make a bit nicer with a nicer error message).

**Comment by strickvl** [comment]
@schustmi ok you're probably right. The confusion (when using it) came when getting an artifact via the `last_successful_run` (where potentially this might have been from a different artifact store and which (when getting the artifact) was calling `load_artifact` under the hood. This flag wouldn't have helped in that case. I'll remove it.

**Comment by schustmi** [comment]
Yes, the flag should maybe be on the `last_succesful_run` method instead, where people can use it to fetch just the runs on the active stack/artifact store?
But then it's probably too far removed from the artifact loading for anyone to realize its use..

**Review by schustmi — APPROVED** [review]

**Review by stefannica — APPROVED** [review]

**Pull request merged**